core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java

```diff
@@ -179,38 +183,55 @@ public void subscribe(Subscriber<? super T> subscriber, Executor executor, boole
     }
 
     private void subscribe0(SubscriptionImpl subscription) {
+        final Subscriber<Object> subscriber = subscription.subscriber();
+        final Executor executor = subscription.executor();
+
         if (!subscriptionUpdater.compareAndSet(this, null, subscription)) {
-            final Subscriber<?> oldSubscriber = this.subscription.subscriber();
-            if (oldSubscriber == AbortingSubscriber.INSTANCE) {
-                throw new IllegalStateException("cannot subscribe to an aborted publisher");
-            } else {
-                throw new IllegalStateException(
-                        "subscribed by other subscriber already: " + oldSubscriber);
-            }
+            failLateSubscriber(executor, subscriber, this.subscription.subscriber());
+            return;
         }
 
-        final Executor executor = subscription.executor();
         if (executor != null) {
             executor.execute(() -> subscription.subscriber().onSubscribe(subscription));
         } else {
-            subscription.subscriber().onSubscribe(subscription);
+            subscriber.onSubscribe(subscription);
```

Comment:
I didn't because that will make the anonymous class created by the lambda expression will have an extra field (i.e. more memory footprint)